src/analysis/special_functions/gamma/beta.lean

```diff
@@ -0,0 +1,559 @@
+  refine tendsto.congr' ((eventually_ne_at_top 0).mp
+    (eventually_of_forall (λ n hn, (Gamma_seq_mul z hn).symm))) (tendsto.mul _ _),
+  { convert tendsto_coe_nat_div_add_at_top (1 - z), ext1 n, rw add_sub_assoc },
+  { have : ↑π / sin (↑π * z) = 1 / (sin (π * z) / π), by field_simp, rw this,
+    refine tendsto_const_nhds.div _ (div_ne_zero hs pi_ne),
+    rw [←tendsto_mul_iff_of_ne_zero tendsto_const_nhds pi_ne, div_mul_cancel _ pi_ne],
+    convert tendsto_euler_sin_prod z,
+    ext1 n, rw [mul_comm, ←mul_assoc] },
+end
+
+/-- The Gamma function does not vanish on `ℂ` (except at non-positive integers, where the function
+is mathematically undefined and we set it to `0` by convention). -/
+theorem Gamma_ne_zero {s : ℂ} (hs : ∀ m : ℕ, s ≠ -m) : Gamma s ≠ 0 :=
+begin
+  by_cases h_im : s.im = 0,
+  { have : s = ↑s.re,
+    { conv_lhs { rw ←complex.re_add_im s }, rw [h_im, of_real_zero, zero_mul, add_zero] },
+    rw [this, Gamma_of_real, of_real_ne_zero],
+    refine real.Gamma_ne_zero (λ n, _),
+    specialize hs n,
+    contrapose! hs,
+    rwa [this, ←of_real_nat_cast, ←of_real_neg, of_real_inj] },
+  { have : sin (↑π * s) ≠ 0,
+    { rw complex.sin_ne_zero_iff,
+      intro k,
+      apply_fun im,
+      rw [of_real_mul_im, ←of_real_int_cast, ←of_real_mul, of_real_im],
+      exact mul_ne_zero real.pi_pos.ne' h_im },
+    have A := div_ne_zero (of_real_ne_zero.mpr real.pi_pos.ne') this,
+    rw [←complex.Gamma_mul_Gamma_one_sub s, mul_ne_zero_iff] at A,
+    exact A.1 }
+end
+
+lemma Gamma_eq_zero_iff (s : ℂ) : Gamma s = 0 ↔ ∃ m : ℕ, s = -m :=
+begin
+  split,
+  { contrapose!, exact Gamma_ne_zero },
+  { rintro ⟨m, rfl⟩, exact Gamma_neg_nat_eq_zero m },
+end
+
+end complex
+
+namespace real
+
+/-- The sequence with `n`-th term `n ^ s * n! / (s * (s + 1) * ... * (s + n))`, for real `s`. We
+will show that this tends to `Γ(s)` as `n → ∞`. -/
+noncomputable def Gamma_seq (s : ℝ) (n : ℕ) :=
+(n : ℝ) ^ s * n! / ∏ (j : ℕ) in finset.range (n + 1), (s + j)
+
+/-- Euler's limit formula for the real Gamma function. -/
+lemma Gamma_seq_tendsto_Gamma (s : ℝ) : tendsto (Gamma_seq s) at_top (𝓝 $ Gamma s) :=
+begin
+  suffices : tendsto (coe ∘ Gamma_seq s : ℕ → ℂ) at_top (𝓝 $ complex.Gamma s),
+    from (complex.continuous_re.tendsto (complex.Gamma ↑s)).comp this,
+  convert complex.Gamma_seq_tendsto_Gamma s,
+  ext1 n,
+  dsimp only [Gamma_seq, function.comp_app, complex.Gamma_seq],
+  push_cast,
+  rw [complex.of_real_cpow n.cast_nonneg, complex.of_real_nat_cast]
+end
+
+/-- Euler's reflection formula for the real Gamma function. -/
+lemma Gamma_mul_Gamma_one_sub (s : ℝ) : Gamma s * Gamma (1 - s) = π / sin (π * s) :=
+begin
+  simp_rw [←complex.of_real_inj, complex.of_real_div, complex.of_real_sin,
+    complex.of_real_mul, ←complex.Gamma_of_real, complex.of_real_sub, complex.of_real_one],
+  exact complex.Gamma_mul_Gamma_one_sub s
+end
+
+end real
+
+end gamma_reflection
+
+section inv_gamma
```

Comment:
This section (up to the end of the file) is new code, not just moving stuff around.